### PR TITLE
Support %I for 12 hour times

### LINKED ISSUE #340: col_date or col_datetime does not support %I hour format for AM/PM tagged hours
For example, parsing date "12/31/1991 6:00:00 PM", given date format  = "%m/%d/%Y %I" will return an error. 

readr::parse_datetime("12/31/1991 6:00 PM", "%m/%d/%Y %I")

### PR DESCRIPTION
**Pull request by jimhester**
Fixes #340

### CODE REVIEW

File: src/DateTimeParser.h [hunk 1]
```diff
@@ -161,6 +161,14 @@ class DateTimeParser {
         if (!consumeInteger(2, &hour_))
           return false;
         break;
+      case 'I': // hour
```
Comment (hadley) [inline src/DateTimeParser.h]: Do we also want to support single digit times?

Comment (jimhester) [inline src/DateTimeParser.h]: This seems to support both single and double digit times.

Comment (hadley) [inline src/DateTimeParser.h]: Ok - can you add a unit test for that case too?

Comment (jimhester) [inline src/DateTimeParser.h]: Done!

**Comment by hadley** [comment]
LGTM

**Pull request merged**